modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,232 @@
+====
+
+.Procedure
+
+. Edit the virtual machine YAML as follows:
++
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> Add the label `special: key` in the `spec.template.metadata.labels` section.
++
+
+[NOTE]
+====
+Labels on a virtual machine are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels in
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
+
+. Save the virtual machine YAML to apply your changes.
+
+. Edit the service YAML to configure the settings necessary to create and expose the Service object.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <3>
+  selector:
+    special: key <4>
+  type: ClusterIP <5>
+----
+<1> Specify the `name` of the service you are creating and exposing.
+<2> Specify `namespace` in the `metadata` section of the Service YAML that corresponds to the `namespace` you specify in the virtual machine YAML.
+<3> Add `targetPort: 22`, exposing the service on SSH port `22`.
+<4> In the `spec` section of the Service YAML, add `special: key` to the `selector` attribute, which corresponds to the `labels` you added in the virtual machine YAML configuration file.
+<5> In the `spec` section of the Service YAML, add `type: ClusterIP` for a
+ClusterIP service. To create and expose other types of services externally, outside of the cluster, such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate.
++
+
+. Save the Service YAML to store the service configuration.
+. Create the `ClusterIP` service.
++
+
+[source,terminal]
+----
+$ oc create -f <service_name>.yaml
+----
+
++
+. Start the virtual machine. If the virtual machine is already running, restart it.
++
+
++
+. Query the Service object to verify it is available and is configured with type `ClusterIP`.
++
+
+.Verification steps
```

Comment:
It seems strange to have a "Vertification steps" title in the middle of the procedure. I would have expected these only to appear _after_ a completed procedure. At least it seems like that was the intended usage per https://redhat-documentation.github.io/modular-docs/#procedure-module-guidelines.

Reply:
Ignoring for now as we discussed.